src/dataflow/components/dataflow-program.tsx

```diff
@@ -48,42 +52,42 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
         new RelayReteNodeFactory(numSocket)];
       if (!this.toolDiv) return;
 
-      const editor = new Rete.NodeEditor(RETE_APP_IDENTIFIER, this.toolDiv);
-      editor.use(ConnectionPlugin);
-      editor.use(ReactRenderPlugin);
-      editor.use(ContextMenuPlugin);
+      this.programEditor = new Rete.NodeEditor(RETE_APP_IDENTIFIER, this.toolDiv);
+      this.programEditor.use(ConnectionPlugin);
+      this.programEditor.use(ReactRenderPlugin);
+      this.programEditor.use(ContextMenuPlugin);
 
       const engine = new Rete.Engine(RETE_APP_IDENTIFIER);
 
       components.map(c => {
-        editor.register(c);
+        this.programEditor.register(c);
         engine.register(c);
       });
 
       const n1 = await components[4].createNode();
       const n2 = await components[0].createNode({ num: 10 });
       const logic = await components[3].createNode();
 
-      n1.position = [80, 200];
-      n2.position = [80, 400];
-      logic.position = [500, 240];
+      n1.position = [80, 80];
+      n2.position = [80, 440];
+      logic.position = [450, 200];
 
-      editor.addNode(n1);
-      editor.addNode(n2);
-      editor.addNode(logic);
+      this.programEditor.addNode(n1);
+      this.programEditor.addNode(n2);
+      this.programEditor.addNode(logic);
 
-      editor.connect(n1.outputs.get("num")!, logic.inputs.get("num1")!);
-      editor.connect(n2.outputs.get("num")!, logic.inputs.get("num2")!);
+      this.programEditor.connect(n1.outputs.get("num")!, logic.inputs.get("num1")!);
+      this.programEditor.connect(n2.outputs.get("num")!, logic.inputs.get("num2")!);
 
-      (editor as any).on(
+      (this.programEditor as any).on(
```

Comment:
Is this cast necessary?

Reply:
This was from the Rete example.  We looked at this briefly with Eli.  Taking it out causes errors.

Reply:
Odd, given that I believe `this.programEditor.on()` is called elsewhere in the same routine without the cast.

Reply:
Struggles with the list of events if the case to `any` is removed.  We can also do this (although I don't see any advantage):
`this.programEditor.on("process nodecreated noderemoved connectioncreated connectionremoved" as any`

Reply:
Sounds like something better TypeScript types would resolve. Leave it for now.